pulp_rpm/app/models/repository.py

```diff
@@ -350,6 +359,20 @@ def finalize_new_version(self, new_version):
                     "{value_errors}"
                 ).format(repo=new_version.repository.name, value_errors=str(ve))
             )
+    
+    def _check_signatures(self, new_version):
+        """If requested, validate that the added rpms are signed appropriately."""
+        if self.allowed_pub_keys:
+            rejected_nevras = []
+            for package in Package.objects.filter(pk__in=new_version.added()).iterator():
+                if str(package.signer_key_id) not in self.allowed_pub_keys:
```

Comment:
I think this line should be 

```if package.signer_key_id is None or str(package.signer_key_id) not in self.allowed_pub_keys```

Otherwise it will try to compare `"None"` which will still work as expected in practice but isn't the best code